### DISCUSSION TITLE
Applying initial conditions to custom power system

### DISCUSSION THREAD

**Discussion by TimThacker** [CURENT/andes#357] [Modeling] [answered]
I am working with an extremely simple power system consisting of a single generator connected to a unit power factor load through a single line. The power flow will not converge for this system if proper initial conditions are not set for the differential and algebraic state variables.

How should I approach defining initial conditions appropriately without having to change model defaults?

**Comment by cuihantao** [comment, marked as answer]
What do "proper initial conditions" specifically mean? What are the model defaults you need to change?

Typically, one needs to specify the initial condition for the power flow so that the iterative solution process can converge. When the power flow solves, the program will then initialize for dynamic analysis such as time-domain simulation.

**Reply by TimThacker** [reply]
I need to change the model defaults for omega, Pm, Pe, V, and Theta. I was hoping there was a way to set initial conditions without changing model defaults in the model code.

In the simple system I am using, I have a classical generator connected to bus 1 set to E∠0 as a reference. The load connected to the generator, a unit power factor impedance load, considers V and Theta. I am looking to create a step change in load impedance to create an under frequency event, to then observe the governor response.

What I have observed while working with this system in MATLAB is that the power flow will converge to a trivial solution without a very specific set of initial conditions.

I have attached a photo of the simple system I am modeling: 
![Simple_Power_sys_diagram](https://user-images.githubusercontent.com/47113711/184561742-5cefbd38-f061-42f8-89ea-0c75c6722e4b.JPG)

**Reply by cuihantao** [reply]
I'm still confused. The solution of power flow will not depend on omega, Pm, or Pe at all. They are initialized after the power flow solution. V and Theta of buses are solved by the power flow program.

In general, dynamic components, including synchronous generators, are initialized after the power flow. In other words, the initial condition for the variables of dynamic models is based on the power flow solution. The program doesn't allow specifying custom `Pm`, for example, since it can lead to inconsistency.

Is it that your power flow won't solve? If so, you might need to adjust the impedance and the load to make it converge first. No need to look at any dynamic model if the power flow won't solve. Please check the single-machine infinite bus system in `andes/cases/smib`.

**Reply by TimThacker** [reply]
I believe the issue lies with there being multiple solutions to power flow and the solver converging to an unintended solution. Thank you for the reference to the SMIB system and the clarification of the system initialization.

**Reply by cuihantao** [reply]
This is possible, and you can tell by checking the voltage magnitudes of the solutions.

To provide a different starting point, change `a0` and `v0` of the buses.
